src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -632,6 +632,32 @@ public async Task EmptyResponse_FrameReceived_ConnectionError()
             }
         }
 
+        [ConditionalFact(nameof(SupportsAlpn))]
+        public async Task CompletedResponse_WindowUpdateFrameReceived_Success()
```

Comment:
Would this test fail without the change in Http2Connection.cs ?

Reply:
Yes. `WaitForConnectionShutdownAsync` throws if any unexpected frames are received on the connection before it shuts down. In this case it would receive a RST_STREAM and throw.

Reply:
Here's the actual failure when I revert the fix:
```
System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.CompletedResponse_WindowUpdateFrameReceived_Success [FAIL]
        System.Exception : Unexpected frame received while waiting for client shutdown: Length: 4
  Type: RstStream
  Flags: None
  StreamId: 1
  Error Code: 5
        Stack Trace:
          C:\corefx\src\Common\tests\System\Net\Http\Http2LoopbackServer.cs(271,0): at System.Net.Test.Common.Http2LoopbackServer.WaitForConnectionShutdownAsync()
          C:\corefx\src\System.Net.Http\tests\FunctionalTests\HttpClientHandlerTest.Http2.cs(657,0): at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http2.CompletedResponse_WindowUpdateFrameReceived_Success()
```